sematic/examples/testing_pipeline/pipeline.py

```diff
@@ -596,8 +644,45 @@ def testing_pipeline(
         futures.append(add_using_resource(initial_future, 1.0))
         futures.append(add_inline_using_resource(initial_future, 1.0))
 
-    if resource_requirements is not None:
-        future = add_with_resource_requirements(initial_future, 3)
+    if expand_shared_memory:
+        k8_resource_requirements = KubernetesResourceRequirements(
+            mount_expanded_shared_memory=True
+        )
+        resource_requirements = ResourceRequirements(
+            kubernetes=k8_resource_requirements
+        )
+
+        future = add_with_expanded_shared_memory(initial_future, 3)
+        future.set(resource_requirements=resource_requirements)
+        futures.append(future)
+
+    if mount_host_paths:
+        k8_host_path_mounts = []
+        pod_mount_paths = []
+        for node_path, pod_mount_path in mount_host_paths:
+            # from the kubernetes documentation:
+            # > a lowercase RFC 1123 label must consist of lower case alphanumeric
+            # > characters or '-', and must start and end with an alphanumeric character
+            # > (e.g. 'my-name',  or '123-abc', regex used for validation is
+            # > '[a-z0-9]([-a-z0-9]*[a-z0-9])?')
+            name = f"volume{pod_mount_path.replace('/', '-')}"
```

Comment:
mebbe default to an auto-generated vol name so the user can't footgun so easily.  what i saw is if the validation fails, the error only appears in the server logs, not in the pod logs nor anything the user is gonna see.  this parameter is really just there to help k8s track the volume and mount objects, it has nothing to do with what the user is prolly focusing on: the host and mount paths.

this is a safe default:
https://github.com/pwais/sematic/blob/e311b10923ae40fa2c070dab5a9cf2f121112c0a/sematic/scheduling/kubernetes.py#L804
`volume_name = f'sematic_host_mount_{str(uuid.uuid4())}'`